tests/testthat/test-child-model.R

```diff
@@ -32,8 +30,158 @@ test_that("Child model can be run for all years", {
+  p_hiv_deaths <- out$p_hiv_deaths[6:15, , ]
+  dimnames(p_hiv_deaths) <- list(age = 5:14, sex = c('male','female'), year = 1970:2030)
+  hc2_df <- as.data.frame(as.table(hc2))
+  out_df <- as.data.frame(as.table(p_hiv_deaths))
+  colnames(hc2_df) <- c("Var1", "Var2", "Var3", "strat")
+  colnames(out_df) <- c("Var1", "Var2", "Var3", "pop")
+  c2 <- hc2_df %>%
+    dplyr::inner_join(out_df, by = c("Var1", "Var2", "Var3")) %>%
+    dplyr::mutate(diff = strat - pop)
+  expect_true(all(abs(c2$diff) < 1e-5))
+
+  ###############################
+  ##Stratified hiv pop and population hiv pop should be the same
+  ###############################
+  ##p_hiv_pop & hc1_hiv_pop, hc1_art_pop, hc2_hiv_pop, hc2_hiv_pop
+  hc1_hiv <- apply(out$hc1_hiv_pop, c(3,4,5), sum)
+  hc1_art <- apply(out$hc1_art_pop, c(3,4,5), sum)
+  hc1 <- hc1_hiv + hc1_art
+  dimnames(hc1) <- list(age = 0:4, sex = c('male','female'), year = 1970:2030)
+  p_hiv <- out$p_hiv_pop[1:5, , ]
+  dimnames(p_hiv) <- list(age = 0:4, sex = c('male','female'), year = 1970:2030)
+  hc1_df <- as.data.frame(as.table(hc1))
+  out_df <- as.data.frame(as.table(p_hiv))
+  colnames(hc1_df) <- c("Var1", "Var2", "Var3", "strat")
+  colnames(out_df) <- c("Var1", "Var2", "Var3", "pop")
+  c1 <- hc1_df %>%
+    dplyr::inner_join(out_df, by = c("Var1", "Var2", "Var3")) %>%
+    dplyr::mutate(diff = strat - pop)
+  expect_true(all(abs(c1$diff) < 1e-5))
+
+  hc2_hiv <- apply(out$hc2_hiv_pop, c(3,4,5), sum)
+  hc2_art <- apply(out$hc2_art_pop, c(3,4,5), sum)
+  hc2 <- hc2_hiv + hc2_art
+  dimnames(hc2) <- list(age = 5:14, sex = c('male','female'), year = 1970:2030)
+  p_hiv <- out$p_hiv_pop[6:15, , ]
+  dimnames(p_hiv) <- list(age = 5:14, sex = c('male','female'), year = 1970:2030)
+  hc2_df <- as.data.frame(as.table(hc2))
+  out_df <- as.data.frame(as.table(p_hiv))
+  colnames(hc2_df) <- c("Var1", "Var2", "Var3", "strat")
+  colnames(out_df) <- c("Var1", "Var2", "Var3", "pop")
+  c2 <- hc2_df %>%
+    dplyr::inner_join(out_df, by = c("Var1", "Var2", "Var3")) %>%
+    dplyr::mutate(diff = strat - pop)
+  expect_true(all(abs(c2$diff) < 1e-5))
+
+})
+
+test_that("Female 15-49y pop aligns", {
+    input <- readRDS(test_path("testdata/child_parms.rds"))
+  demp <- input$demp
+  parameters <- input$parameters
+  dp <- input$dp
+  pjnz <- input$pjnz
+
+  out <- run_model(demp, parameters, 1970:2030, NULL, run_child_model = TRUE)
+  spec <- SpectrumUtils::dp.output.hivpop(dp, direction = 'long')
+  spec <- spec %>%
+    dplyr::filter(Age %in% 15:49 & Sex == 'Female') %>%
+    dplyr::group_by(Year)
+
+  lfrog <- out$p_hiv_pop[16:50,2,]
+  dimnames(lfrog) <- list(Age = 15:49, Year = 1970:2030)
+  lfrog <- as.data.frame(as.table(lfrog)) %>%
+     dplyr::mutate(lfrog = Freq)
+  lfrog$Year <- as.integer(as.character(lfrog$Year))
+
+
+  dt <- dplyr::right_join(spec, lfrog, by = c("Year", "Age"))
+  dt <- dt %>%
+    dplyr::mutate(diff = Value - lfrog)
+
+ # expect_true(all(abs(dt$diff) < 1e-3))
```

Comment:
Is this not working atm, or can we uncomment it?

If test is not working currently, but you expect it to in the future, prefer a `testthat::skip("reason why I am skipping this test")` at the top of the function. With the explicit skip we'll see it when tests are run so it's easier to remember to turn it on down the line.

Reply:
Added a skip comment.